api/tests/integration/infrastructure/repositories/user-repository_test.js

```diff
@@ -5,233 +5,351 @@ const bcrypt = require('bcrypt');
+              knex('users').delete(),
+              knex('organization-roles').delete()
+            ]);
+          });
       });
-    });
 
-    it('should return organization access associated to the user', () => {
-      // when
-      const promise = userRepository.findByEmailWithRoles(email);
+      it('should return user informations for the given email', () => {
+        // given
+        const expectedUser = new User(userInDB);
+
+        // when
+        const promise = userRepository.findByEmailWithRoles(userInDB.email);
+
+        // then
+        return promise.then((user) => {
+          expect(user).to.be.an.instanceof(User);
+          expect(user.id).to.equal(expectedUser.id);
+          expect(user.firstName).to.equal(expectedUser.firstName);
+          expect(user.lastName).to.equal(expectedUser.lastName);
+          expect(user.email).to.equal(expectedUser.email);
+          expect(user.password).to.equal(expectedUser.password);
+          expect(user.cgu).to.equal(expectedUser.cgu);
+        });
+      });
 
-      // then
-      return promise.then((user) => {
-
-        expect(user.organizationsAccesses).to.be.an('array');
-
-        const firstOrganizationAccess = user.organizationsAccesses[0];
-        expect(firstOrganizationAccess).to.be.an.instanceof(OrganizationAccess);
-        expect(firstOrganizationAccess.id).to.equal(organizationAccess.id);
-
-        const accessibleOrganization = firstOrganizationAccess.organization;
-        expect(accessibleOrganization).to.be.an.instanceof(Organization);
-        expect(accessibleOrganization.id).to.equal(organization.id);
-        expect(accessibleOrganization.code).to.equal(organization.code);
-        expect(accessibleOrganization.name).to.equal(organization.name);
-        expect(accessibleOrganization.type).to.equal(organization.type);
-        expect(accessibleOrganization.email).to.equal(organization.email);
-
-        const associatedRole = firstOrganizationAccess.organizationRole;
-        expect(associatedRole).to.be.an.instanceof(OrganizationRole);
-        expect(associatedRole.id).to.equal(organizationRole.id);
-        expect(associatedRole.name).to.equal(organizationRole.name);
+      it('should return organization access associated to the user', () => {
+        // when
+        const promise = userRepository.findByEmailWithRoles(userToInsert.email);
+
+        // then
+        return promise.then((user) => {
+
+          expect(user.organizationsAccesses).to.be.an('array');
+
+          const firstOrganizationAccess = user.organizationsAccesses[0];
+          expect(firstOrganizationAccess).to.be.an.instanceof(OrganizationAccess);
+          expect(firstOrganizationAccess.id).to.equal(organizationAccessInDB.id);
+
+          const accessibleOrganization = firstOrganizationAccess.organization;
+          expect(accessibleOrganization).to.be.an.instanceof(Organization);
+          expect(accessibleOrganization.id).to.equal(organizationInDB.id);
```

Comment:
Possibilité de passer par :  expect(accessibleOrganization).to.deep.equal(organizationInDB) ?